simpleclient/src/main/java/io/prometheus/client/SimpleCollector.java

```diff
@@ -174,12 +176,42 @@ protected SimpleCollector(Builder b) {
     for (String n: labelNames) {
       checkMetricLabelName(n);
     }
+    labelValueSanitizer = b.labelValueSanitizer;
 
     if (!b.dontInitializeNoLabelsChild) {
       initializeNoLabelsChild();
     }
   }
 
+  /**
+   * Default Sanitizer - Will not perform any manipulation on labels
+   */
+  static LabelValueSanitizer NoOpLabelValueSanitizer() {
+    return new LabelValueSanitizer() {
+      @Override
+      public String[] sanitize(String... labelValue) {
+        return labelValue;
+      }
+    };
+  }
+
+  /**
+   * Transforms null labels to an empty string to guard against IllegalArgument runtime exceptions
+   */
+  static LabelValueSanitizer TransformNullLabelsToEmptyString() {
+    return new LabelValueSanitizer() {
+      @Override
+      public String[] sanitize(String... labelValue) {
+        for (int i = 0; i < labelValue.length; i++) {
+          if (labelValue[i] == null) {
+            labelValue[i] = "";
+          }
+        }
+        return labelValue;
+      }
+    };
+  }
+
```

Comment:
If the sanitizer takes only a single string as argument instead of an array of strings, a good handler for null values would be `String::valueOf`. As this already exists in the Java runtime, there's no need to provide our own implementation in `SimpleCollector`.